cni-plugins/to_docker/README.md

```diff
@@ -0,0 +1,112 @@
+a container.  In particular, this plugin's "add container to network"
+operation connects the container to a Docker network that is
+identified in the config file.
+
+To meet all of the functional requirements of
+http://kubernetes.io/docs/admin/networking/#kubernetes-model some
+additional static configuration is required, which you must do
+yourself.
+
+This plugin is written in bash so that is widely understandable and
+does not require any building.
+
+
+## Pre-requisites
+
+You must have Kubernetes, Docker, bash, and [jq]
+(https://stedolan.github.io/jq/) installed on each Kubernetes worker
+machine ("node").
+
+Docker must be configured with a cluster store, and you must have a
+multi-host Docker network.  The Docker name of that multi-host network
+must appear as the value of the `name` field in `c2d.conf`.
+
+For example, you might use the Kuryr libnetwork driver to make a
+multi-host Docker network with the following command.
+
+```
+docker network create -d kuryr --ipam-driver=kuryr --subnet=172.19.0.0/16 --gateway=172.19.0.1 mynet
+```
+
+For an almost-example, you might try Docker's "overlay" driver to make
+a multi-host Docker network with the following command.
+
+```
+docker network create -d overlay --subnet=172.19.0.0/16 --gateway=172.19.0.1 mynet
+```
+
+However, with Docker's "overlay" driver there is no way to add the
+required additional connectivity from hosts to containers.  So do not
+do this.
+
+For another almost-example, you might try Calico's libnetwork driver,
+with commands like the following.
+
+```
+calicoctl pool add 172.19.0.0/16 --nat-outgoing
+calicoctl pool remove 192.168.0.0/16
+docker network create --driver calico --ipam-driver calico mynet
+```
+
+However, Calico's libnetwork driver gives the name `cali0` to the
+network interface inside the container --- which conflicts with
+Kubernetes' requirement that this network interface be named `eth0`.
+So do not do this.  Perhaps [eventually]
+(https://github.com/appc/cni/issues/113) CNI will allow the network
+interface name to be returned, which would enable Kubernetes to cope
+with variant names.
+
+
+## Installation and Configuration
+
+This plugin has one config file and two executables.  Put the
+executables in `/opt/cni/bin/`.  Put the config file, `c2d.conf`, in a
+directory of your choice; `/etc/cni/net.d/` is the usual choice.
+
+There are two configuration settings that must be made on each
+kubelet.  The following expresses those settings as command line
+arguments, assuming that `/etc/cni/net.d/` is the directory where you
+put the config file.
+
+```
+--network-plugin=cni --network-plugin-dir=/etc/cni/net.d
+```
+
+If the config file has a field named `debug` with the value `true`
+then each invocation of the plugin will produce some debugging output
+in `/tmp/`.
+
+A multi-host docker network does not necessarily meet the requirement
+(seen in http://kubernetes.io/docs/admin/networking/#kubernetes-model)
```

Comment:
I mean part of "all nodes can communicate with all containers (and vice-versa) without NAT".  I say *part of* because, as far as I know, the vice-versa is not really required (except as logically required by the use of NodePort).